container/container.go

```diff
@@ -723,12 +723,20 @@ func (container *Container) CreateDaemonEnvironment(tty bool, linkedEnv []string
 	if os == "" {
 		os = runtime.GOOS
 	}
-	env := []string{}
+
+	// Figure out what size slice we need so we can allocate this all at once.
+	envSize := len(container.Config.Env)
 	if runtime.GOOS != "windows" || (runtime.GOOS == "windows" && os == "linux") {
-		env = []string{
-			"PATH=" + system.DefaultPathEnv(os),
-			"HOSTNAME=" + container.Config.Hostname,
-		}
+		envSize += 2 + len(linkedEnv)
+	}
+	if tty {
+		envSize++
+	}
+
+	env := make([]string, 0, envSize)
+	if runtime.GOOS != "windows" || (runtime.GOOS == "windows" && os == "linux") {
```

Comment:
Looking at this "if", I think we currently only support "windows" or "linux" here, so effectively this whole "if" is `if os == "linux"` ?

Reply:
It's "not windows or lcow", so not exactly `os == "linux"`

Reply:
If the only OSes we have are Linux and Windows, then;

- `runtime.GOOS != "windows"` means we're running on "linux" (and implicitly, OS of the container is Linux)
- `runtime.GOOS == "windows" && os == "linux"` means the container is Linux

Alternatively, `if os != "windows"` ?

Reply:
Officially we only support linux (as far as not windows), but in practice things are setup for other OSes as well.